File: C++/maximum-number-of-operations-with-the-same-score-i.cpp
// Time:  O(n)
// Space: O(1)

// array
class Solution {
public:
    int maxOperations(vector<int>& nums) {
        int result = 1;
        const int target = nums[0] + nums[1];
        for (int i = 2; i + 1 < size(nums); i += 2) {
            if (nums[i] + nums[i + 1] != target) {
                break;
            }
            ++result;
        }
        return result;
    }
};

File: Python/maximum-number-of-operations-with-the-same-score-i.py
# Time:  O(n)
# Space: O(1)

# array
class Solution(object):
    def maxOperations(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        result = 1
        target = nums[0]+nums[1]
        for i in xrange(2, len(nums)-1, 2):
            if nums[i]+nums[i+1] != target:
                break
            result += 1
        return result
